builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -346,6 +357,18 @@ func (p *Provisioner) deployConfigFiles(
 		return fmt.Errorf("Uploading %s failed: %v", validationKey, err)
 	}
 
+        // Open the secret key file
+        s, err := os.Open(p.SecretKeyPath)
+        if err != nil {
+                return err
+        }
+        defer s.Close()
+
+        // Copy the secret key to the new instance
+        if err := comm.Upload(path.Join(confDir, secretKey), s); err != nil {
+                return fmt.Errorf("Uploading %s failed: %v", secretKey, err)
+	}
+
```

Comment:
We should first check if `p.SecretKeyPath != ""` so that we only do this when we actually gave a secret key